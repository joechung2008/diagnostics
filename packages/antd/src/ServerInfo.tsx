import { Table } from "antd";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  const columns = [
    {
      title: "Name",
      dataIndex: "name",
      key: "name",
    },
    {
      title: "Value",
      dataIndex: "value",
      key: "value",
    },
  ];

  return (
    <Table
      aria-label="Server Info"
      columns={columns}
      dataSource={items.map((item) => ({ ...item, key: item.name }))}
      pagination={false}
    />
  );
};

export default ServerInfo;
