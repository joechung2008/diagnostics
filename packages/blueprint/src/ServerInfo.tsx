import { HTMLTable } from "@blueprintjs/core";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <HTMLTable aria-label="Server Info">
      <thead>
        <tr>
          <th>Name</th>
          <th>Value</th>
        </tr>
      </thead>
      <tbody>
        {items.map((item) => (
          <tr key={item.name}>
            <td>{item.name}</td>
            <td>{item.value}</td>
          </tr>
        ))}
      </tbody>
    </HTMLTable>
  );
};

export default ServerInfo;
