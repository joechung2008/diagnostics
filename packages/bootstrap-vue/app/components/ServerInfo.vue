<template>
  <div>
    <BTable :items="rows" :fields="fields" small />
  </div>
</template>

<script setup lang="ts">
import type { ServerInfoProps } from "@/types/ServerInfoProps";
import { when } from "@/utils/extensions";

const props = defineProps<ServerInfoProps>();

const fields = [
  { key: "name", label: "Name" },
  { key: "value", label: "Value" }
];

const rows = computed(() => [
  { name: "Hostname", value: props.hostname },
  ...when(props.uptime !== undefined, {
    name: "Uptime",
    value: String(props.uptime)
  }),
  { name: "Server ID", value: props.serverId },
  { name: "Deployment ID", value: props.deploymentId },
  ...when(props.nodeVersions !== undefined, {
    name: "Node Versions",
    value: props.nodeVersions
  }),
  {
    name: "Extension Sync | Total Sync All Count",
    value: String(props.extensionSync.totalSyncAllCount)
  }
]);
</script>
