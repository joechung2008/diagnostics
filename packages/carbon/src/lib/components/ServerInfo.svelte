<script lang="ts">
	import { DataTable } from 'carbon-components-svelte';

	const { deploymentId, extensionSync, hostname, nodeVersions, serverId, uptime }: ServerInfoProps =
		$props();

	const headers = [
		{ key: 'name', value: 'Name', empty: false },
		{ key: 'value', value: 'Value', empty: false }
	];

	const rows = $derived([
		{ id: 'hostname', name: 'Hostname', value: hostname },
		{ id: 'uptime', name: 'Uptime', value: uptime },
		{ id: 'server-id', name: 'Server ID', value: serverId },
		{ id: 'deployment-id', name: 'Deployment ID', value: deploymentId },
		{ id: 'node-versions', name: 'Node Versions', value: nodeVersions },
		{
			id: 'extension-sync',
			name: 'Extension Sync | Total Sync All Count',
			value: extensionSync.totalSyncAllCount
		}
	]);
</script>

<DataTable aria-label="Server Information" {headers} {rows} />
