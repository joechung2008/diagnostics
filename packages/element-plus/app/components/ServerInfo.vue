<template>
    <el-table :data="rows" style="width: 100%" aria-label="Server Info">
        <el-table-column prop="name" label="Name" show-overflow-tooltip />
        <el-table-column prop="value" label="Value" show-overflow-tooltip />
    </el-table>
</template>

<script setup lang="ts">
import type { ServerInfoProps } from "@/types"

const props = defineProps<ServerInfoProps>()

const rows = computed(() => [
    { name: "Hostname", value: props.hostname },
    { name: "Uptime", value: props.uptime },
    { name: "Server ID", value: props.serverId },
    { name: "Deployment ID", value: props.deploymentId },
    { name: "Node Versions", value: props.nodeVersions },
    {
        name: "Extension Sync | Total Sync All Count",
        value: props.extensionSync.totalSyncAllCount
    }
])
</script>
