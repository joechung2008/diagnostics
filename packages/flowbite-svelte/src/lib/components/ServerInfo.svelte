<script lang="ts">
  import {
    Table,
    TableBody,
    TableBodyCell,
    TableBodyRow,
    TableHead,
    TableHeadCell
  } from "flowbite-svelte";

  let { deploymentId, extensionSync, hostname, nodeVersions, serverId, uptime }: ServerInfoProps =
    $props();

  const items = $derived.by(() => [
    { name: "Hostname", value: hostname },
    { name: "Uptime", value: uptime },
    { name: "Server ID", value: serverId },
    { name: "Deployment ID", value: deploymentId },
    { name: "Node Versions", value: nodeVersions },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount
    }
  ]);
</script>

<Table aria-label="Server Info">
  <TableHead>
    <TableHeadCell>Name</TableHeadCell>
    <TableHeadCell>Value</TableHeadCell>
  </TableHead>
  <TableBody>
    {#each items as item (item.name)}
      <TableBodyRow>
        <TableBodyCell>{item.name}</TableBodyCell>
        <TableBodyCell>{item.value}</TableBodyCell>
      </TableBodyRow>
    {/each}
  </TableBody>
</Table>
