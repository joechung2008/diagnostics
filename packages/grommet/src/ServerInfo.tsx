import { Table, TableBody, TableCell, TableHeader, TableRow } from "grommet";

const ServerInfo: React.FC<ServerInfoProps> = ({
  deploymentId,
  extensionSync,
  hostname,
  nodeVersions,
  serverId,
  uptime,
}) => {
  const items = [
    {
      name: "Hostname",
      value: hostname,
    },
    {
      name: "Uptime",
      value: uptime,
    },
    {
      name: "Server ID",
      value: serverId,
    },
    {
      name: "Deployment ID",
      value: deploymentId,
    },
    {
      name: "Node Versions",
      value: nodeVersions,
    },
    {
      name: "Extension Sync | Total Sync All Count",
      value: extensionSync.totalSyncAllCount,
    },
  ];

  return (
    <Table aria-label="Server Info">
      <TableHeader>
        <TableRow>
          <TableCell scope="col">Name</TableCell>
          <TableCell scope="col">Value</TableCell>
        </TableRow>
      </TableHeader>
      <TableBody>
        {items.map((item) => (
          <TableRow key={item.name}>
            <TableCell>{item.name}</TableCell>
            <TableCell>{item.value}</TableCell>
          </TableRow>
        ))}
      </TableBody>
    </Table>
  );
};

export default ServerInfo;
