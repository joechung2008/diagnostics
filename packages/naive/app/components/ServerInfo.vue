<template>
  <n-data-table :columns="columns" :data="rows" size="small" />
</template>

<script setup lang="ts">
import type { ServerInfoProps } from "@/types/ServerInfoProps"
import { when } from "~/utils/extensions"

const props = defineProps<ServerInfoProps>()

const columns = [
  { title: "Name", key: "name" },
  { title: "Value", key: "value" }
]

const rows = computed(() => [
  { name: "Hostname", value: props.hostname },
  ...when(props.uptime !== undefined, {
    name: "Uptime",
    value: String(props.uptime)
  }),
  { name: "Server ID", value: props.serverId },
  { name: "Deployment ID", value: props.deploymentId },
  ...when(props.nodeVersions !== undefined, {
    name: "Node Versions",
    value: props.nodeVersions
  }),
  {
    name: "Extension Sync | Total Sync All Count",
    value: String(props.extensionSync.totalSyncAllCount)
  }
])
</script>
