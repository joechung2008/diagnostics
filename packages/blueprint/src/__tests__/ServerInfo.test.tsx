import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import ServerInfo from "../ServerInfo";

describe("ServerInfo", () => {
  it("matches snapshot", () => {
    const mockProps = {
      deploymentId: "deploy123",
      extensionSync: { totalSyncAllCount: 0 },
      hostname: "localhost",
      nodeVersions: "",
      os: "Windows",
      serverVersion: "1.0.0",
      serverId: "server123",
      uptime: 1000,
    };
    const { asFragment } = render(<ServerInfo {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
