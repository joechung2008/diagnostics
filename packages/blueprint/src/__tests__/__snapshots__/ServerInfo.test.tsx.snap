// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServerInfo > matches snapshot 1`] = `
<DocumentFragment>
  <table
    aria-label="Server Info"
    class="bp6-html-table"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Hostname
        </td>
        <td>
          localhost
        </td>
      </tr>
      <tr>
        <td>
          Uptime
        </td>
        <td>
          1000
        </td>
      </tr>
      <tr>
        <td>
          Server ID
        </td>
        <td>
          server123
        </td>
      </tr>
      <tr>
        <td>
          Deployment ID
        </td>
        <td>
          deploy123
        </td>
      </tr>
      <tr>
        <td>
          Node Versions
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Extension Sync | Total Sync All Count
        </td>
        <td>
          0
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
